test/clj/matcher_combinators/matchers_test.clj

```diff
@@ -148,3 +148,32 @@
                 ::result/value {:actual   b
                                 :expected a}
                 ::result/weight 1}))))
+
+(defrecord RecordA [val])
+(defrecord RecordB [val])
+
+(facts "records"
```

Comment:
would be nice to add some tests in `midje-test` namespace just to see how this change plays out in practice